src/clean.c

```diff
@@ -288,14 +299,22 @@ static int clean_staged_manifests(const char *path, bool dry_run, bool all)
 			continue;
 		}
 
+		/* Delta manifests are only valid when updating FROM the current version
+		 * TO the new version and should not be cached. */
+		string_or_die(&version_dir, "%s/%s", state_dir, name);
```

Comment:
`version_dir` gets reassigned to the same thing again before `remove_if(version_dir, dry_run, is_manifest)` below. Remove the reassignment down there.

Reply:
This also changes a bit in #574 which should be merged soon, so this will need an update.